pkg/collector/corechecks/cluster/orchestrator/processors/processor_test.go

```diff
@@ -43,3 +45,33 @@ func TestChunkItems(t *testing.T) {
 	actual := chunkResources(items, 3, 2)
 	assert.ElementsMatch(t, expected, actual)
 }
+
+func TestSortedMarshal(t *testing.T) {
+	p := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-pod",
+			Annotations: map[string]string{
+				"b-annotation":   "test",
+				"ab-annotation":  "test",
+				"a-annotation":   "test",
+				"ac-annotation":  "test",
+				"ba-annotation":  "test",
+				"1ab-annotation": "test",
+			},
+		},
+	}
+	yaml, err := json.Marshal(p)
+	assert.NoError(t, err)
+
+	/*	Expected order should be :
```

Comment:
- its expected json, no? 
- why not just reformatted the var to reflect the comment? This way we don't need the comment

Reply:
That makes sense, I rename to `json` and use `JSONEQ` instead to avoid redundant comment
[aa63182](https://github.com/DataDog/datadog-agent/pull/13386/commits/aa63182f6790775105a517b8ce6f8f7e92874d16)